packages/react-langgraph/src/useLangGraphMessages.ts

```
@@ -32,10 +34,18 @@ export const useLangGraphMessages = <TMessage>({
 
   const sendMessage = useCallback(
     async (newMessages: TMessage[], config: LangGraphSendMessageConfig) => {
-      const optimisticMessages = [...messages, ...newMessages];
-      if (newMessages.length > 0) {
-        setMessages(optimisticMessages);
-      }
+      // ensure all messages have an ID
+      newMessages = newMessages.map((m) => (m.id ? m : { ...m, id: uuidv4() }));
```

Biome (1.9.4)

[error] Reassigning a function parameter is confusing.

The parameter is declared here:

Use a local variable instead.

(lint/style/noParameterAssign)